File: 08/nameTester.rb
# NameTester.rb tests class Name and its operations
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: David Yoo (gy24)
# Date: Mar 15, 2022
####################################################

require 'test/unit/assertions'   # needed for assert
include Test::Unit::Assertions

# Replace this line with the declaration of class Name
class Name
   def initialize (first, middle, last)
      @first, @middle, @last = first, middle, last
   end
   attr_reader :first, :middle, :last
   def fullName
      @first + " " + @middle + " " + @last
   end
   def print
      puts fullName
      fullName
   end
   def lfmi
    return @last + ", " + @first + " " + @middle[0]
   end
   def readName
    @first = gets.chomp
    @middle = gets.chomp
    @last = gets.chomp
   end
   attr_accessor :first, :middle, :last
end


def testName
  name = Name.new("John", "Paul", "Jones")

  assert name.first == "John", "first failed"
  assert name.middle == "Paul", "middle failed"
  assert name.last == "Jones", "last failed"
  assert name.fullName == "John Paul Jones", "fullName failed"
  assert name.print == "John Paul Jones", "print failed"

  assert name.first = "David"
  assert name.first == "David", "setting first name failed"
  assert name.middle = "Gunju"
  assert name.middle == "Gunju", "setting middle name failed"
  assert name.last = "Yoo"
  assert name.last == "Yoo", "setting last name failed"

  assert name.lfmi == "Yoo, David G", "lfmi failed"
   
  print "All tests passed!\n"
  print "Please enter your first, middle, last name and press enter each time you enter your name):"
  name.readName
  puts name.fullName
end

testName

File: projects/08/NameTester.java
/* NameTester.java is a 'driver' to test class Name.
 * Author: David Yoo (gy24)
 * Date: Mar 15, 2022
 ************************************************************/

import java.util.Scanner;
public class NameTester  
{
  public static void main(String[] args)
  {
     Name aName = new Name("John", "Paul", "Jones");

     assert aName.getFirst().equals("John") ;
     assert aName.getMiddle().equals("Paul") ;
     assert aName.getLast().equals("Jones") ;

     System.out.println(aName);
     assert aName.toString().equals("John Paul Jones");

     aName.setFirst("David");
     assert aName.getFirst().equals("David");
	   aName.setMiddle("Gunju");
	   assert aName.getMiddle().equals("Gunju");
	   aName.setLast("Yoo");
	   assert aName.getLast().equals("Yoo");

	   assert aName.lfmi().equals("Yoo, David G");

     System.out.println("All tests passed!");
     System.out.println("Please enter your first, middle, last name and press enter each time you enter your name):");
	   aName.readName();
	   System.out.println(aName);
  }
}

// Replace this line with the declaration of class Name...
class Name
{
  private String first, middle, last;
  public Name(String first, String middle, String last){
    this.first = first;
    this.middle = middle;
    this.last = last;
  }
  public String getFirst(){
    return first;
  }
  public String getMiddle(){
    return middle;
  }
  public String getLast(){
    return last;
  }
  public String toString(){
    return first + ' ' + middle + ' ' + last;
  }
  public void setFirst(String first){
    this.first = first;
  }
  public void setMiddle(String middle){
    this.middle = middle;
  }
  public void setLast(String last){
    this.last = last;
  }
  String lfmi() {
    return last + ", " + first + ' ' + middle.charAt(0);
  }
  private Scanner scanner = new Scanner(System.in);

    
 void readName() {
    first = scanner.nextLine();
	  middle = scanner.nextLine();
	  last = scanner.nextLine();
 }
}
